src/App.tsx

```diff
@@ -116,75 +126,97 @@ const hexRegex = /^#[a-f0-9]{6}$/gi;
 
 // TODO: add an error boundary for parse fails
 const Editor: Component<{ currentFile: FileSystemFileHandle }> = (props) => {
-  const [contents, { mutate }] = createResource(async () => {
+  const [obj, setObject] = createStore<Record<string, unknown>>({});
+  let lastUpdate = Date.now();
+  createEffect(async () => {
     const file = await props.currentFile.getFile();
     const text = await file.text();
-    console.log(parseJsonFile(text));
-    return text;
+    const json = JSON.parse(text);
+    setObject(reconcile(json));
+  });
+
+  createEffect(() => {
+    const interval = setInterval(async () => {
+      const file = await props.currentFile.getFile();
+      const text = await file.text();
+      const json = JSON.parse(text);
+
+      if (lastUpdate < file.lastModified) {
+        setObject(reconcile(json));
+        lastUpdate = Date.now();
+      }
+    }, 1000);
+    return () => clearInterval(interval);
```

Comment:
```suggestion
    onCleanup(() => clearInterval(interval));
```

Reply:
ooh, good catch! committed that to main now since that's an important fix